doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+These are called _Root Keys_ (based on Unix's filesystem root, as well as the root of a tree).
+
+Lets explore them one by one:
+
+- The simplest Namespace is the **"Meta"**.
+  The Namespace "Meta" is used exclusively for Meta Keys, i.e. Keys that are attached to another Key as metadata.
+  The Key Hierarchy of the Namespace "Meta" is entirely separate from the Key Hierarchies in the other Namespaces.
+  Without external information, you cannot determine from a Key Name with Namespace "Meta", which Key this Meta Key is attached to.
+- Keys with Namespace **"Proc"** only exist in memory and are scoped to the current <b>Proc</b>ess.
+- Keys with Namespace **"Dir"** are scoped to a single filesystem <b>Dir</b>ectory.
+  They will (normally) be stored somewhere within this directory.
+  Currently, there is no way of knowing which filesystem directory a Key with Namespace "Dir" is scoped to.
+  Elektra only uses Keys scoped to the Current Working Directory.
+- Similarly, Keys with Namespace **"User"** are scoped to a single **User**.
+  They will (normally) be stored somewhere within the User's home directory.
+  Again, there is no way of knowing which user a Key is scoped to and Elektra only uses Keys scoped to the current user, i.e. the one that executed the application.
+- The Namespace **"System"** is what makes Elektra global.
+  Keys with Namespace "System" are the same for all users of the system, independent of context.
+  They are stored in system level directory.
+- Keys with Namespace "Default" are special.
+  While you could create them manually, you normally don't want to.
+  It is used for Keys with default values for Namespace Resolution (explained below).
+  Keys with this Namespace are also in-memory only.
+- Then there is **"Spec"**.
+  This Namespace, like "Meta", is separate from the rest.
+  We use it for Keys that are part of a specification used to describe other Keys.
+  The Metadata of every Key with Namespace "Spec" describes a specification for the Keys that have the same Key Name Part but a different Namespace (except "Meta").
+  So the Namespace "Spec" has a closer relation to the others than "Meta".
+- Now just the Namespace **"Cascading"** remains.
+  This Namespace is used for Namespace Resolution (see below).
+  It is the one that applications and end-users will use most commonly when interacting with Elektra.
+  Keys with Namespace "Cascading" are never stored.
+  Not on disk and ideally also not in a `KeySet`.
+
+There is also a certain ranking between the Namespaces "Proc", "Dir", "User", "System" and "Default".
+Namely, that they override each other in exactly this order.
+Given two Key Names with identical Key Name Parts, but one with Namespace "Dir" and one with Namespace "User", the one with Namespace "Dir" should be considered more specific and should be preferred.
+
+A special feature of Elektra is _Namespace Resolution_ (sometimes just Name Resolution, or Key Name Resolution).
```

Comment:
Please use only one term and stick to it.

Reply:
I did stick to "Namespace Resolution" in this document, but other places (e.g. doc/tutorials/namespaces.md) talk about "resolving the (key) name". This want I wanted to indicate. I will use a different phrase to clarify that "resolving the name" == "resolving the namespace".

Reply:
But I can't look through all of the documentation to find every mention of "resolving a keyname" or something like that. I chose "Namespace Resolution", because it is more accurate to what I describe here. As footnote 2 indicates, the whole process used by `ksLookup` involves a bit more.

If you think about it this distinction makes sense. The Namespace Resolution is the process of finding the correct namespace. This is just a part of Name Resolution which finds the whole key name and therefore takes into account things like `override` metakeys.

Reply:
Why not simply call it (cascading) lookup?

To not call it "resolving" is actually a good idea as "resolving" is associated with the file name resolver plugins.

Btw. there is a single occurrence of "resolved" (as verb!) in doc/tutorials/namespaces.md, so I would not call this an "established term".